rs/registry/canister/src/mutations/do_update_subnet.rs

```diff
@@ -250,27 +250,23 @@ impl Registry {
     }
 
     /// Validates that the SEV feature is not changed on an existing subnet.
+    ///
     /// Panics if the SEV feature is attempted to be changed.
     fn validate_update_sev_feature(&self, payload: &UpdateSubnetPayload) {
-        if payload.features.is_none() {
-            return;
-        }
         let subnet_id = payload.subnet_id;
-        let subnet_record = self.get_subnet_or_panic(subnet_id);
-        if let Some(old_features) = subnet_record.features {
-            // Compare as `SubnetFeatures`, to avoid having to worry about
-            // `None` vs `Some(false)`.
-            let new_features: SubnetFeatures = payload.features.unwrap().into();
-            let old_features: SubnetFeatures = old_features.into();
-            if new_features.sev_enabled == old_features.sev_enabled {
-                return;
-            }
+
+        // Ensure the subnet record exists for this subnet ID.
+        let _subnet_record = self.get_subnet_or_panic(subnet_id);
+
+        let new_sev_feature_flag = payload.features.and_then(|features| features.sev_enabled);
+
+        if let Some(new_sev_feature_flag) = new_sev_feature_flag {
```

Comment:
NCR.

I guess this replaces

```
            if new_features.sev_enabled == old_features.sev_enabled {
```

(but has the opposite sense). Why is the old value not used here? IIUC, all we are doing is saying that you are not allowed to change sev_enabled. I guess what you are doing is a little bit stronger: you also forbid setting sev_enabled to its current value. I guess that is ok, but it seems like you have changed the spec behavior, not just made the implementation match the spec.

Reply:
> you also forbid setting sev_enabled to its current value

Yes, I decided to simplify the validation code by making it stricter in a way that to me makes more sense.